spec/classes/foreman_proxy_content_spec.rb

```diff
@@ -6,36 +6,48 @@
       let(:facts) { facts }
 
       context 'without parameters' do
-        it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_package('katello-debug') }
-      end
-
-      context 'with pulp' do
-        let(:params) do
-          {
-            qpid_router: false
-          }
-        end
-
         let(:pre_condition) do
           <<-PUPPET
           include foreman_proxy
           class { 'foreman_proxy::plugin::pulp':
             enabled          => false,
-            pulpnode_enabled => true,
+            pulpnode_enabled => false,
           }
           PUPPET
         end
 
         it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_class('pulp').with(manage_squid: true) }
-        it { is_expected.not_to contain_class('foreman_proxy_content::dispatch_router') }
-        it { is_expected.not_to contain_class('pulpcore') }
-        it { is_expected.to contain_class('foreman_proxy_content::pub_dir') }
+        it { is_expected.to contain_package('katello-debug') }
+      end
 
-        it do
-          is_expected.to contain_pulp__apache__fragment('gpg_key_proxy')
-            .with_ssl_content(%r{ProxyPass /katello/api/v2/repositories/ https://foo\.example\.com/katello/api/v2/repositories/})
+      if facts[:operatingsystemmajrelease] == '7'
+        context 'with pulp' do
```

Comment:
Like in the first comment, you can write less indented code by using `if:` here
```suggestion
      context 'with pulp', if: facts[:operatingsystemmajrelease] == '7' do
```